### ISSUE TITLE
Q: Incorrect neural network dimensionality?

### ISSUE THREAD

**Issue by giladturok** [senya-ashukha/real-nvp-pytorch#1]
I think the input dimension of the scale and translate neural networks $s$ and $t$ may be incorrect.

The original RealNVP [paper](https://arxiv.org/pdf/1605.08803) states that the dimensionality of networks $s$ and $t$ are $R^d \rightarrow R^D-d$ for some $d < D$. In this code, the data (moons and normal) is $D=2$. However, the code defines the input layers of networks $s$ and $t$ as $d=2$ instead of $d=1 < D$. I believe this is a mistake that has not been detected despite the popularity of this repo.

Screenshot from this code example:

<img width="653" alt="Screenshot 2024-09-26 at 8 23 17 PM" src="https://github.com/user-attachments/assets/74d5da04-3845-4253-9d01-9dc56fa6e0a3">

Screenshot from the paper:

<img width="1276" alt="Screenshot 2024-09-26 at 8 03 25 PM" src="https://github.com/user-attachments/assets/1593d480-4e38-48d1-aa1b-0fb7bd558777">

This mistake is easy to make because a later equation in the paper suggests that the input dimension of networks $s$ and $t$ should be $D$ instead of some $d < D$. In the screenshot below, the terms $s(b \cdot x)$ and $t(b \cdot x)$ are misleading because although $b \cdot x \in R^D$, we actually want to pass in *only* the non-masked elements of $x$ (which is in $R^d$).

<img width="1279" alt="Screenshot 2024-09-26 at 8 05 45 PM" src="https://github.com/user-attachments/assets/0a2af0c4-a080-431a-8fe1-ded3ca1f20b6">

If I'm making a mistake, please let me know!
